socs/agents/stm_encoder/drivers.py

```diff
@@ -0,0 +1,283 @@
+#!/usr/bin/env python3
+'''Module to read stimulator encoder.
+'''
+import fcntl
+import mmap
+import os
+import sys
+from pathlib import Path
+from queue import Queue
+from threading import Thread
+from time import sleep
+
+import numpy as np
+
+ADDR_AXI = 0x80020000
+PATH_DEV_BASE = Path(f'/sys/devices/platform/axi/{ADDR_AXI:08x}.str_rd/uio')
+PATH_LOCK = Path(__file__).parent.joinpath('.lock')
```

Comment:
This puts the lock file in where the socs module is installed, which isn't necessarily writable. This should be moved somewhere else. I would suggest `/tmp/`.